contracts/interfaces/IFractalUsul.sol

```diff
@@ -2,44 +2,166 @@
 pragma solidity ^0.8.0;
 
 import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+import "../FractalUsul.sol";
 
 interface IFractalUsul {
-    /// @dev Returns the hash of a transaction in a proposal.
-    /// @param proposalId the proposal to inspect.
-    /// @param index the transaction to inspect.
-    /// @return transaction hash.
-    function getTxHash(uint256 proposalId, uint256 index)
-        external
-        view
-        returns (bytes32);
+    enum ProposalState {
+        Active,
+        Canceled,
+        Timelocked,
+        Executed,
+        Executable,
+        Uninitialized
```

Comment:
in what instance would a proposal be `Uninitialized`?  A user would need to submit a proposal when there is no strategy enabled?

I'm just wondering if there's a way we can refactor that state out and handle not getting to it with `require` checks instead

Reply:
Yeah, it would only be `Uninitialized` when you query a proposal ID that hasn't been submitted yet. 

I agree this state isn't necessary, and should revert instead, so I just pushed a commit to move this into a require statement check, and removed that `Uninitialized` state from the enum.

Reply:
awesome